stable/yugabyte/templates/_helpers.tpl

```diff
@@ -3,26 +3,64 @@
 Expand the name of the chart.
 */}}
 {{- define "yugabyte.name" -}}
-{{- default .Chart.Name .Values.nameOverride | trunc 63 | trimSuffix "-" -}}
+{{- default .Chart.Name .Values.nameOverride | trunc 43 | trimSuffix "-" -}}
 {{- end -}}
 
 {{/*
 Create a default fully qualified app name.
 We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
 If release name contains chart name it will be used as a full name.
+The components in this chart create additional resources that expand the longest created name strings.
+The longest name that gets created of 20 characters, so truncation should be 63-20=43.
 */}}
 {{- define "yugabyte.fullname" -}}
 {{- if .Values.fullnameOverride -}}
-{{- .Values.fullnameOverride | trunc 63 | trimSuffix "-" -}}
+{{- .Values.fullnameOverride | trunc 43 | trimSuffix "-" -}}
 {{- else -}}
 {{- $name := default .Chart.Name .Values.nameOverride -}}
 {{- if contains $name .Release.Name -}}
-{{- .Release.Name | trunc 63 | trimSuffix "-" -}}
+{{- .Release.Name | trunc 43 | trimSuffix "-" -}}
 {{- else -}}
-{{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" -}}
+{{- printf "%s-%s" .Release.Name $name | trunc 43 | trimSuffix "-" -}}
 {{- end -}}
 {{- end -}}
 {{- end -}}
+
+{{/* Generate common labels */}}
+{{- define "yugabyte.labels" }}
+heritage: {{ .Values.helm2Legacy | ternary "Tiller" (.Release.Service | quote) }}
+release: {{ .Release.Name | quote }}
+chart: {{ .Values.oldNamingStyle | ternary .Chart.Name (include "yugabyte.chart" .) | quote }}
+component: {{ .Values.Component | quote }}
+{{- if .Values.commonLabels}}
+{{ toYaml .Values.commonLabels }}
+{{- end }}
+{{- end }}
+
+{{/* Generate app label */}}
+{{- define "yugabyte.applabel" }}
+{{- if .root.Values.oldNamingStyle }}
+app: "{{ .label }}"
+{{- else }}
+app.kubernetes.io/name: "{{ .label }}"
```

Comment:
Is there any reason why we wouldn't want to switch to using the recommended app name label all the time? ref: https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/

CC: @Arnav15, @iSignal